File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
      int n = nums.size();
      if (n == 0) return 0;
      int current = nums[0];
      int maxcount = 1;
      int count = 1;
      sort(nums.begin(),nums.end());
      for(int i = 0 ; i < n ; i++){
        if(nums[i]==current+1){
            count++;
            current=  nums[i];
        }else if(nums[i]==current){
            continue;
        }else{
            count = 1;
            current = nums[i];
        }
        maxcount = max(maxcount,count);
      
      }
      return maxcount;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        if(nums.length ==0) return 0;
        int maxi = 1;
        int count = 1;
        int current = nums[0];
        Arrays.sort(nums);
        for(int i = 0 ; i < nums.length ; i++){
            if(nums[i]==current+1){
                count++;
                current= nums[i];
                maxi = Math.max(maxi,count);
            }else if(nums[i]==current){
                continue;
            }else{
                count = 1;
                current = nums[i];
            }
            // maxi = Math.max(count,maxi);
        }
        return maxi;
    }
}